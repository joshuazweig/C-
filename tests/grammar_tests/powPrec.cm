int main() {
    int x;

    x = 6 * 2 ^ 2 / 8;
}

