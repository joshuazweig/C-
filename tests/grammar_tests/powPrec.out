ACCEPT
[program:
  [decls:
    [decls:]
    [fdecl:
      [typ: INT]
      ID
      LPAREN
      [formals_opt:]
      RPAREN
      LBRACE
      [vdecl_list: [vdecl_list:] [vdecl: [typ: INT] ID SEMI]]
      [stmt_list:
        [stmt_list:]
        [stmt:
          [expr:
            ID
            ASSIGN
            [expr:
              [expr:
                [expr: LITERAL]
                STAR
                [expr: [expr: LITERAL] POW [expr: LITERAL]]
              ]
              DIVIDE
              [expr: LITERAL]
            ]
          ]
          SEMI
        ]
      ]
      RBRACE
    ]
  ]
  EOF
]
